tmail-backend/rate-limiter/rate-limiter-api/src/main/scala/com/linagora/tmail/rate/limiter/api/RateLimitation.scala

```diff
@@ -0,0 +1,97 @@
+package com.linagora.tmail.rate.limiter.api
+
+import com.linagora.tmail.rate.limiter.api.LimitTypes.LimitTypes
+import eu.timepit.refined
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.collection.NonEmpty
+import org.apache.james.rate.limiter.api.AllowedQuantity.AllowedQuantity
+
+import java.time.Duration
+import java.util.UUID
+import scala.util.Try
+
+object LimitTypes {
+  type LimitTypes = Seq[LimitType] Refined NonEmpty
+
+  def validate(value: Seq[LimitType]): Either[IllegalArgumentException, LimitTypes] =
+    refined.refineV[NonEmpty](value) match {
+      case Right(value) => Right(value)
+      case Left(_) => Left(new IllegalArgumentException("value should not be empty"))
+    }
+
+  def liftOrThrow(value: Seq[LimitType]): LimitTypes =
+    validate(value) match {
+      case Right(value) => value
+      case Left(error) => throw error
+    }
+}
+
+sealed trait LimitType {
+  def asString(): String
+
+  def allowedQuantity(): AllowedQuantity
+}
+
+case class Count(quantity: AllowedQuantity) extends LimitType {
+  override def asString(): String = "count"
+
+  override def allowedQuantity(): AllowedQuantity = quantity
+}
+
+case class Size(quantity: AllowedQuantity) extends LimitType {
+  override def asString(): String = "size"
+
+  override def allowedQuantity(): AllowedQuantity = quantity
+}
+
+case class RateLimitation(name: String, period: Duration, limits: LimitTypes)
+
+object RateLimitingPlanId {
+  def generate: RateLimitingPlanId = RateLimitingPlanId(UUID.randomUUID())
+
+  def from(value: String): Try[RateLimitingPlanId] = Try(RateLimitingPlanId(UUID.fromString(value)))
+}
+
+case class RateLimitingPlanId(value: UUID)
+
+case class RateLimitingPlanName(value: String)
+
+
+sealed trait TransitionType {
+  def asString(): String
+
+  def rateLimitations(): Seq[RateLimitation]
+}
```

Comment:
How about `OperationLimitations` that could be `TransitLimitations`, `RelayLimitations` or `DeliveryLimitations` ?